src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -305,6 +306,27 @@ public boolean apply(StepDescriptor i) {
         return r;
     }
 
+    /** Return true if we can easily create a nice String for user display from the object */
+    static boolean isAbleToUseToStringForDisplay(@CheckForNull Object o) {
+        return (o != null && (o instanceof Comparable || o instanceof CharSequence)); // Covers our base types
```

Comment:
Well how often do you expect a null single value parameter? Would seem very rare that there is a step taking a single parameter and nobody provides a value for it. So I suspect the 95% case will be non-null and you hit the chain anyway... smells like premature optimisation.

As to useful toStrings... we want implenters to actually override the method in the 60% of cases. Much easier to implement a strict whitelist and extend case by case when there is evidence of general safety... you are exposing the general first and when you lock down then plugin authors have a "breaking" change.

Make your SPI strict to start and only open up as you need to. So I recommend locking this stuff *down* as this is an SPI